src/content/hinting.ts

```diff
@@ -270,18 +343,48 @@ class Hint {
         if (config.get("hintuppercase") == "true") {
             this.flag.classList.add("TridactylHintUppercase")
         }
-        /* this.flag.style.cssText = ` */
-        /*     top: ${rect.top}px; */
-        /*     left: ${rect.left}px; */
-        /* ` */
-        this.flag.style.cssText = `
-            top: ${window.scrollY + offsetTop + rect.top}px !important;
-            left: ${window.scrollX + offsetLeft + rect.left}px !important;
-        `
+        this.y = this.originY = window.scrollY + offsetTop + rect.top
+        this.x = this.originX = window.scrollX + offsetLeft + rect.left
         modeState.hintHost.appendChild(this.flag)
         this.hidden = false
     }
 
+    set x(X: number) {
+        this._x = X
+        this.updatePosition()
+    }
+
+    get x() {
+        return this._x
+    }
+
+    set y(Y: number) {
+        this._y = Y
+        this.updatePosition()
+    }
+
+    get y() {
+        return this._y
+    }
+
+    private updatePosition() {
+        this.flag.style.cssText = `
+            top: ${this._y}px !important;
+            left: ${this._x}px !important;
+        `
+    }
+
+    public overlapsWith(h: Hint) {
+        if (this.width == 0) this.width = this.flag.getClientRects()[0].width
```

Comment:
I tried a simpler approach by copy & paste some of the code of this pull. And realised here the `h.width/h.height` could be `0` in the `overlapsWith()` call. If that happened, overlap would not be detected.